File: ALifeCommon/src/main/java/com/cristiano/alife/instructions/branch/RetInstruction.java
package com.cristiano.alife.instructions.branch;

import com.cristiano.alife.consts.ALifeConsts;
import com.cristiano.alife.world.BaseInstruction;
import com.cristiano.alife.world.IOrganismo;
import com.cristiano.alife.world.IWorld;

public class RetInstruction extends BaseInstruction  {
	
	public RetInstruction(IWorld mundo) {
		super(mundo);
	}
	
	@Override
	public void executa(IOrganismo o) {
		int pos=o.pop();
		o.setIp(pos-1);
	}
	
	@Override
	public String getDescription(IOrganismo o,int ip){
		return comment(name,"ip <- pop()");
	}


}

File: netAvida/netAvida.Backend/instructions/impl/branch/RetInstruction.cs
using netAvida.backend;
using netAvida.backend.interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace netAvida.Backend.instructions.impl
{
    public class RetInstruction : BaseInstruction
    {
        public RetInstruction(IWorld mundo): base(mundo)
        {
        }

    public override void executa(IOrganismo o)
        {
            int pos = o.pop();
            o.ip=pos - 1;
        }


    public override String getDescription(IOrganismo o, int ip)
        {
            return comment(name, "ip <- pop()");
        }
    }
}
